src/snowflake/snowpark/types.py

```diff
@@ -333,10 +334,16 @@ class ArrayType(DataType):
     def __init__(
         self,
         element_type: Optional[DataType] = None,
-        structured: bool = False,
+        structured: Optional[bool] = None,
     ) -> None:
-        self.structured = structured
-        self.element_type = element_type if element_type else StringType()
+        if context._should_use_structured_type_semantics:
+            self.structured = (
+                structured if structured is not None else element_type is not None
+            )
+            self.element_type = element_type
+        else:
```

Comment:
can `element_type` be `None` here? What does it mean for the column type to be so?

Reply:
If element_type is None than this is a semi-structured array column and could contain anything.